middleware/url_format_test.go

```diff
@@ -67,3 +69,30 @@ func TestURLFormatInSubRouter(t *testing.T) {
 		t.Fatalf(resp)
 	}
 }
+
+func TestURLFormatWithoutChiRouteContext(t *testing.T) {
+	r := chi.NewRouter()
+
+	r.Use(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			newCtx := context.WithValue(r.Context(), chi.RouteCtxKey, nil)
```

Comment:
Hi @Taaipi,

Thanks for the PR!

It looks like this test cases injects a `nil` RouteContext down the chain intentionally.

Do you have a scenario how could this happen in real-world use case? I wonder how is the `middleware.URLFormat` ever called without chi route context unless we set it to `nil` explicitly.

Thanks!

Reply:
I guess based on https://github.com/go-chi/chi/pull/718, it'd be better to create a test case with some sub-routers (instead of setting `nil`). Right?

Reply:
Are you using middleware.URLFormat outside of chi router?